### PR TITLE
Remove `BackgroundColorProvider` from Halstack

### PR DESCRIPTION
**Pull request by GomezIvann**
**Checklist**

- [x] The Build process is done without errors. All tests pass in the `/lib` directory.
- [x] Self-reviewed the code before submitting.
- [x] Meets accessibility standards.
- [x] Added/updated documentation to `/website` as needed.
- [x] Added/updated tests as needed.

**Description**
Our design tokens are currently under review, and our future way of implementing the Dark Theme will likely be very different from the current one. Until we figure out how to properly build a hierarchy of tokens. This functionality is going to be in the shadows.

### CODE REVIEW

File: lib/src/spinner/Spinner.tsx [hunk 1]
```diff
@@ -109,13 +73,14 @@ const DxcSpinner = ({
   );
 };
 
-const determinatedValue = (value: SpinnerPropsType["value"], strokeDashArray:number) => {
+const determinedValue = (value: SpinnerPropsType["value"], strokeDashArray: number) => {
```
Comment (Jialecl) [inline lib/src/spinner/Spinner.tsx:76]: I think the correct word determinate. Determine means to "figure out".

File: lib/src/switch/Switch.tsx [hunk 2]
```diff
@@ -125,65 +110,53 @@ const calculateWidth = (margin, size) =>
...
-const getDisabledColor = (props, element, subelement?) => {
+const getDisabledColor = (
+  theme: AdvancedTheme["switch"],
+  element: "track" | "thumb" | "label",
+  subElement?: "check" | "uncheck"
+) => {
   switch (element) {
     case "track":
-      switch (subelement) {
+      switch (subElement) {
```
Comment (Jialecl) [inline lib/src/switch/Switch.tsx:120]: Subelement is a single word.

File: lib/src/text-input/TextInput.tsx [hunk 3]
```diff
@@ -786,67 +673,53 @@ const Action = styled.button<{ backgroundType: BackgroundColors }>`
...
+    const color = props.disabled ? props.theme.disabledSuffixColor : props.theme.suffixColor;
     return `color: ${color}; border-left: 1px solid ${color};`;
   }};
   font-family: ${(props) => props.theme.fontFamily};
   font-size: 1rem;
   pointer-events: none;
 `;
 
-const ErrorIcon = styled.span<{ backgroundType: BackgroundColors }>`
+const ErrorIcon = styled.span<{}>`
```
Comment (Jialecl) [inline lib/src/text-input/TextInput.tsx:704]: `<{}>` can be deleted.

**Comment by Jialecl** [comment]
There is a visual test for the container that can be removed since it is testing the container with a black background.

**Comment by Jialecl** [comment]
We still have references to onDark in the advanced theme example and the advanced theme generator.
Should we consider removing those in this PR?

File: app/src/pages/DateInput.jsx [hunk 4]
```diff
@@ -158,20 +157,6 @@ function App() {
           margin={{ left: "medium" }}
         ></DxcButton>
       </div>
-      <BackgroundColorProvider color="#000000">
-        <DarkMode>
```
[the remark this answers is not in the document]
Comment (GomezIvann) [reply app/src/pages/DateInput.jsx:162]: Yes, done!

File: app/src/pages/Button.jsx [hunk 5]
```diff
@@ -556,140 +555,6 @@ function App() {
           </div>
         </HalstackProvider>
       </div>
-      <BackgroundColorProvider color="#000000">
-        <Mode mode="dark" text="Default">
```
[the remark this answers is not in the document]
Comment (GomezIvann) [reply app/src/pages/Button.jsx:560]: Done!

File: app/src/pages/ProgressBar.jsx [hunk 6]
```diff
@@ -126,25 +125,6 @@ function App() {
           />
         </HalstackProvider>
       </div>
-      <h4>Dark mode</h4>
-      <BackgroundColorProvider color="#000000">
-        <Mode mode="dark" text="Undeterminate default">
```
[the remark this answers is not in the document]
Comment (GomezIvann) [reply app/src/pages/ProgressBar.jsx:131]: Done!

File: app/src/pages/Textarea.jsx [hunk 7]
```diff
@@ -204,45 +203,6 @@ function App() {
           margin={{ left: "medium" }}
         ></DxcButton>
       </div>
-      <BackgroundColorProvider color="#000000">
-        <Mode mode="dark">
```
[the remark this answers is not in the document]
Comment (GomezIvann) [reply app/src/pages/Textarea.jsx:208]: Done!

File: lib/src/spinner/Spinner.tsx (hunk 1 above)
Comment (GomezIvann) [reply lib/src/spinner/Spinner.tsx:76]: But, determinated is an adjective: "to be determinated". A certain value is a determined value.

https://www.linguee.es/ingles-espanol/traduccion/determined.html
https://www.linguee.es/ingles-espanol/traduccion/to+be+determinated.html

File: lib/src/switch/Switch.tsx (hunk 2 above)
Comment (GomezIvann) [reply lib/src/switch/Switch.tsx:120]: **subelement** is not a word, is **sub-element**.

File: lib/src/text-input/TextInput.tsx (hunk 3 above)
Comment (GomezIvann) [reply lib/src/text-input/TextInput.tsx:704]: Done!

File: lib/src/spinner/Spinner.tsx (hunk 1 above)
Comment (Jialecl) [reply lib/src/spinner/Spinner.tsx:76]: I checked other design sytems and it seems that the word used is determinate.

https://carbondesignsystem.com/components/progress-bar/usage/
https://carbondesignsystem.com/components/progress-bar/usage/

Comment (GomezIvann) [reply lib/src/spinner/Spinner.tsx:76]: Updated.

**Review by Jialecl — APPROVED** [review]

**Pull request merged**